File: problems/HackerRank/gradingStudents.ts
function gradingStudents(grades: number[]): number[] {
  let result: number[] = [];

  for (let ele of grades) {
    if (ele < 38 || ele % 5 < 3) {
      result.push(ele);
      continue;
    }

    const newGrade = ele + (5 - (ele % 5));
    result.push(newGrade);
  }

  return result;
}

console.log(gradingStudents([33, 38, 40, 57, 58, 84]));

File: problems/HackerRank/week2/gradingStudents.py
def gradingStudents(grades):
    ans = []
    for g in grades:
        if g < 38:
            ans.append(g)
            continue
        
        remainder = g % 5
        if remainder >= 3:
            compliment = (5 - remainder) % 5
            new_grade = g + compliment
            ans.append(new_grade)
        else:
            ans.append(g)
            
    return ans